Lib/configparser.py

```diff
@@ -1106,10 +1106,12 @@ def _handle_rest(self, st, line, fpname):
         # a section header or option header?
         if self._allow_unnamed_section and st.cursect is None:
             st.sectname = UNNAMED_SECTION
-            st.cursect = self._dict()
-            self._sections[st.sectname] = st.cursect
-            self._proxies[st.sectname] = SectionProxy(self, st.sectname)
-            st.elements_added.add(st.sectname)
```

Comment:
I notice now that these four lines are essentially a copy-paste of lines 1134-1137. We should consolidate those too, to avoid this logic drifting. I'll plan to do that before or after this PR.